invoicer_project/static/js/add_edit_item.js

```diff
@@ -0,0 +1,167 @@
+    if (numOfRows >= maxNumOfUnits) {
+        alert(`You can't add more than ${maxNumOfUnits} additional units!`);
+        return;
+    }
+    removeAllErrorAttributes(additionalFieldsContainer);
+    removeAllErrorAttributes(amountAdditionalFieldsContainer);
+    let index = 0;
+    for (let i = 0; i <= numOfRows; i++) {
+        if (additionalUnitCell[i].classList.contains("hidden")) {
+            index = i;
+            break;
+        }
+    }
+    additionalUnitCell[index].classList.remove("hidden");
+    additionalUnits[index].parentNode.classList.remove("d-none");
+    additionalUnits[index].parentNode.classList.add("d-flex");
+    ++numOfRows;
+}
+
+function removeLabels(index) {
+    additionalUnitCell[index].classList.add("hidden");
+    additionalUnits[index].parentNode.classList.add("d-none");
+    additionalUnits[index].parentNode.classList.remove("d-flex");
+    removeTextFromTable(document.querySelector(`#Aditional_unit_${index + 1}`), `Aditional unit ${index + 1}`);
+    removeTextFromTable(document.querySelector(`#AU${index + 1}_val`), "");
+    additionalUnits[index].value = "";
+    amountAdditionalUnitField[index].value = "";
+    --numOfRows;
+}
+
+const closeButton = document.querySelectorAll(".close_btn");
+for (let i=0; i<closeButton.length; i++) {
+    closeButton[i].addEventListener('click', ()=>{removeLabels(i)});
+}
+
+document.querySelector("#additional_item_button").addEventListener("click", () => { addLabels() });
+
+function setTextToTable(inputElement, inputData) {
+    inputElement.setAttribute('data-text', inputData);
+    inputElement.innerHTML = inputElement.getAttribute('data-text');
+}
+
+function removeTextFromTable(inputElement, data) {
+    inputElement.removeAttribute('data-text');
+    inputElement.innerHTML = data;
+}
+
+function isFieldEmpty(input_field, fieldLabel, resultValue) {
+    let data = input_field.value;
+    if (data === fieldLabel) {
+        data = resultValue;
+    }
+    return data;
+}
+
+priceField.addEventListener('input', () => { setTextToTable(inputPriceVal, priceField.value) });
+
+function input_currency_table() {
+    let data = isFieldEmpty(inputCurrency, "Currency", "");
+    setTextToTable(inputCurrencyVal, data);
+}
+
+
+amountInStockField.addEventListener('input', () => {
+    let data = isFieldEmpty(inputBasicUnit, "Basic unit", "");
+    setTextToTable(inputAmountInStockVal, amountInStockField.value + " " + data);
+});
+
+nameField.addEventListener('input', () => {
+    let data = isFieldEmpty(nameField, "", "Name of item");
+    setTextToTable(inputNameVal, data);
+});
+
+barcodeField.addEventListener('input', () => { setTextToTable(inputBarcodeVal, `Barcode: ${barcodeField.value}`) });
+
+
+function input_basic_unit_table() {
+    let data1 = isFieldEmpty(inputBasicUnit, "Basic unit", "");
+
+    setTextToTable(inputBasicUnitVal, data1);
+    setTextToTable(inputAmountInStockVal, amountInStockField.value + " " + data1);
+
+    if (numOfRows > 0) {
```

Comment:
let's reduce count of nested code
```suggestion
    if (numOfRows <= 0) return;
```